src/editor/css_sort.rs

```diff
@@ -0,0 +1,103 @@
+use scraper::Selector;
+use serde::Deserialize;
+use url::Url;
+
+use super::Editor;
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct CssSort {
+    #[serde(deserialize_with = "super::deserialize_selector")]
+    pub selector: Selector,
+
+    #[serde(default)]
+    pub reverse: bool,
+
+    #[serde(default, skip_serializing_if = "Vec::is_empty")]
+    pub sort_by: Vec<Editor>,
+}
+
+impl CssSort {
+    pub fn apply(&self, url: &Url, html: &str) -> anyhow::Result<String> {
+        let parsed_html = scraper::Html::parse_document(html);
+        let mut selected = parsed_html.select(&self.selector).collect::<Vec<_>>();
+
+        if selected.is_empty() {
+            anyhow::bail!("selected nothing");
+        }
+
+        selected.sort_by_cached_key(|item| {
+            let mut content = super::Content {
+                extension: Some("html"),
+                text: item.html(),
+            };
+            for editor in &self.sort_by {
+                if let Ok(inner) = editor.apply(url, content) {
+                    content = inner;
+                } else {
+                    return String::new();
+                }
+            }
+            content.text
+        });
+
+        if self.reverse {
+            selected.reverse();
+        }
+
+        Ok(selected
+            .iter()
+            .map(scraper::ElementRef::html)
+            .collect::<Vec<_>>()
+            .join("\n"))
+    }
+}
+
+#[test]
+fn simple_example() {
+    let url = Url::parse("https://edjopato.de/").unwrap();
+    let input = "<html><head></head><body><p>A</p><p>C</p><p>B</p></body></html>";
+    let expected = "<p>A</p>
+<p>B</p>
+<p>C</p>";
+    let example = CssSort {
+        selector: Selector::parse("p").unwrap(),
+        sort_by: Vec::new(),
+        reverse: false,
+    };
+    let html = example.apply(&url, input).unwrap();
+    assert_eq!(html, expected);
+}
+
+#[test]
+fn simple_example_reverse() {
+    let url = Url::parse("https://edjopato.de/").unwrap();
+    let input = "<html><head></head><body><p>A</p><p>C</p><p>B</p></body></html>";
+    let expected = "<p>C</p>
+<p>B</p>
+<p>A</p>";
+    let example = CssSort {
+        selector: Selector::parse("p").unwrap(),
+        sort_by: Vec::new(),
+        reverse: true,
+    };
+    let html = example.apply(&url, input).unwrap();
+    assert_eq!(html, expected);
+}
+
```

Comment:
```suggestion

#[test]
fn sorted_selections_remain_sorted() {
...
}

#[test]
fn unsortable_selection_maintains_original_order() {
...
     let input = r#"<html><head></head><body>
<article><h3>C</h3><a id="Yes">Bla</a></article>
<article><h3>A</h3><a id="Yes">Bla</a></article>
</body></html>"#;
    let expected = r#"<article><h3>C</h3><a id="Yes">Bla</a></article>
<article><h3>A</h3><a id="Yes">Bla</a></article>"#;
 let example = CssSort {
        selector: Selector::parse("article").unwrap(),
        sort_by: vec![Editor::CssSelect(Selector::parse("a").unwrap())],
        reverse: false,
    };
    ...
}

#[test]
fn numbers_sorted_before_alphabetically() {
... // this test is basically an extension of the readme, 
    // showing a detail of interest in the behavior
}

#[test]
fn use_css_sort_within_css_sort () {
...
  let input = "
<body>
  <p id="A">
    <div>1. Trains</div>
    <div>2. Plains</div>
  </p>
  <p id="C">
    <div>2. Plants</div>
    <div>1. Stones</div>
  </p>
  <p id="B">
    <div>2. Blue</div>
    <div>1. Red</div>
  </p>
</body>";

    let expected = "
  <p id="A">
    <div>1. Trains</div>
    <div>2. Plains</div>
  </p>
  <p id="B">
    <div>1. Red</div>
    <div>2. Blue</div>
  </p>
  <p id="C">
    <div>1. Stones</div>
    <div>2. Plants</div>
  </p>";
...
}
```

Reply:
> fn numbers_sorted_before_alphabetically

Ah, now that I know about [css selectors](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_selectors) it probably does not make sense to re-explain the already (externally) specified behavior. I still think it might be useful as a quick example in the tests for people but I would put less emphasize on this remark.

Reply:
`use_css_sort_within_css_sort` is currently not possible as it both selects and sorts.

without selecting it would be hard to keep track of the origins and what to actually sort and what not.
`css_sort: div` would need to keep track of where something is from.

This could be done by only sorting `div` inside the same parent :thinking:

Reply:
In my mental state, I wasn't aware that this is not possible. Maybe add a "limitations" section to the README?

Reply:
Separated selecting and sorting: 081186991931b70a9e57cb47c12c02d6f0737ce6

Reply:
Yeah, it should, but I think this rather obvious. It's mainly there to notify the user about "you configured something fishy". This was more relevant when this was both sort and select, now I'm not completely sure whether this should be only a warning instead?

Reply:
I think it should just be a warning. Lets imagine we track the sorted output of an API. One day, there are no items left because the api says so. What would happen? What would we / the user want to see in such an event?

Reply:
`css_select` also errors in that case. And I somewhat like that behavior. When the list is empty, its not ideal, but the more common change is a different website layout resulting in different css selectors. The old ones are wrong now. Its fishy when there is nothing anymore.

With css_select the empty list is relatively easy to handle by using the parent item.
With css_sort this is not the case as it needs to take the given elements to sort… so maybe a warning seems more relevant here. css_select should likely be used before that anyway.

Reply:
On the other hand: the things that are sorted are relevant. When there is nothing to sort, the relevant things are not there. I think I would continue with an error. It can still be changed to a warning later on when a good use case comes up.

Reply:
Exactly my thinking. If I, the user, `css_select` something and its gone the other day: I have to act. If I sort a list of pizzas for sale and one day no pizza is for sale: I do not have to act, eventually there will be pizza once more!

Does that make sense? Lists can be empty but tags should be present.